### PR TITLE
Now transpiles for estimator and sampler

### PR DESCRIPTION
**Pull request by austingmhuang**
Without transpilation, you can't run Estimators and Samplers using Qiskit 1.0. This is a simple change that just transpiles circuits so that it can be submitted to a real hardware backend. Tested on a local notebook that access to the device works as intended and the answers seem reasonable (other than some formatting issues that a different PR is addressing).

### CODE REVIEW

**Comment by austingmhuang** [comment]
[sc-62928]

**Comment by austingmhuang** [comment]
Can hold off on this one while we decide on if we are going to change to using V2 primitives instead.

File: pennylane_qiskit/qiskit_device2.py [hunk 1]
```diff
@@ -623,18 +624,21 @@ def _execute_sampler(self, circuit, session):
     def _execute_estimator(self, circuit, session):
         # the Estimator primitive takes care of diagonalization and measurements itself,
         # so diagonalizing gates and measurements are not included in the circuit
-        qcirc = circuit_to_qiskit(circuit, self.num_wires, diagonalize=False, measure=False)
+        qcirc = [circuit_to_qiskit(circuit, self.num_wires, diagonalize=False, measure=False)]
         if circuit.shots:
             self.options.execution.shots = circuit.shots.total_shots
         estimator = Estimator(session=session, options=self.options)
+        compiled_circuits = self.compile_circuits(qcirc)
```
Comment (obliviateandsurrender) [inline pennylane_qiskit/qiskit_device2.py:631]: I think I have asked this before, but to confirm that this is not the PR that we are compiling based on the device? 🤔

Comment (austingmhuang) [reply pennylane_qiskit/qiskit_device2.py:631]: I think I gave you the wrong answer last time 😅 . 

When you run compile_circuits, it compiles based on the device backend.

Comment (austingmhuang) [reply pennylane_qiskit/qiskit_device2.py:631]: In further detail:

I was confused and thought we didn't transpile when `use_primitives=False`. But actually we were transpiling when `use_primitives=False` and **not** transpiling when` use_primitives = True`. This PR just makes it so when `use_primitives=True`, we transpile.

**Comment by austingmhuang** [comment]
This is just a few lines of code so I merged it into #539 as without the transpilation none of the code works 😅

**Pull request closed without merging**